src/org/traccar/api/resource/AttributeAliasResource.java

```
@@ -0,0 +1,93 @@
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.traccar.Context;
+import org.traccar.api.BaseResource;
+import org.traccar.model.AttributeAlias;
+
+@Path("devices/aliases")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class AttributeAliasResource extends BaseResource {
+
+    @GET
+    public Collection<AttributeAlias> get(@QueryParam("deviceId") long deviceId) throws SQLException {
+        if (deviceId != 0) {
+            if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+                Context.getPermissionsManager().checkDevice(getUserId(), deviceId);
+            }
+            return Context.getAliasesManager().getDeviceAttributeAliases(deviceId);
+        } else {
+            return Context.getAliasesManager().getUserDevicesAttributeAliases(getUserId());
+        }
+    }
+
+    @POST
+    public Response add(AttributeAlias entity) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+            Context.getPermissionsManager().checkDevice(getUserId(), entity.getDeviceId());
+        }
+        Context.getAliasesManager().addAttributeAlias(entity);
+        return Response.ok(entity).build();
+    }
+
+    @Path("{id}")
+    @PUT
+    public Response update(@PathParam("id") long id, AttributeAlias entity) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (!Context.getPermissionsManager().isAdmin(getUserId())) {
+            AttributeAlias oldAttrbuteAlias = Context.getAliasesManager().getAttributeAliasById(entity.getId());
+            if (oldAttrbuteAlias != null && oldAttrbuteAlias.getDeviceId() != entity.getDeviceId()) {
```

Same here. I don't think we need to check. Just let it crash if old doesn't exist. I also think we can also not check if device id is the same, just check both. Less code.